server/threads/queries.ts

```diff
@@ -242,3 +242,24 @@ export const moveThread = async ({
     return false;
   }
 };
+
+export const deleteThread = async ({
+  threadExternalId,
+  // implement mod permissions later
+  firebaseId, 
+}: {
+  threadExternalId: string;
+  firebaseId: string;
+}) => {
+  try {
+    const result = await pool.none(sql.deleteThread, {
```

Comment:
You want to use `pool.tx` here (see: https://github.com/BobaBoard/boba-backend/blob/a7069e90d3fc7a016ccba87811baac11cc8e851d/server/boards/queries.ts#L201)

The reason is that if the query fails midway (e.g. could delete all the tags, but failed on the comments), you want the whole thing to deletion to rollback and the db to be restored to the original state.

See documentation too: https://github.com/vitaly-t/pg-promise#transactions